src/vm/finalizerthread.cpp

```diff
@@ -463,7 +472,9 @@ void FinalizerThread::WaitForFinalizerEvent (CLREvent *event)
             cEventsForWait,                           // # objects to wait on
             &(MHandles[uiEventIndexOffsetForWait]),   // array of objects to wait on
             FALSE,          // bWaitAll == FALSE, so wait for first signal
-#if defined(__linux__) && defined(FEATURE_EVENT_TRACE)
+#if defined (LOWMEMORY_CHECK_TIMEOUT)
```

Comment:
Since we want this to be always enabled, only the LOWMEMORY_CHECK_TIMEOUT needs to be here and we need to add a counter that would ensure that we trigger the heap dump only with the original frequency.